core/src/main/java/org/elasticsearch/bootstrap/BootstrapCLIParser.java

```diff
@@ -0,0 +1,168 @@
+ * under the License.
+ */
+
+package org.elasticsearch.bootstrap;
+
+import com.google.common.base.Strings;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.elasticsearch.Build;
+import org.elasticsearch.common.cli.CliTool;
+import org.elasticsearch.common.cli.CliToolConfig;
+import org.elasticsearch.common.cli.Terminal;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.monitor.jvm.JvmInfo;
+
+import java.util.Iterator;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.elasticsearch.common.cli.CliToolConfig.Builder.cmd;
+import static org.elasticsearch.common.cli.CliToolConfig.Builder.optionBuilder;
+
+public class BootstrapCLIParser extends CliTool {
+
+    private static final CliToolConfig CONFIG = CliToolConfig.config("elasticsearch", BootstrapCLIParser.class)
+            .cmds(Start.CMD, Version.CMD)
+            .build();
+
+    public BootstrapCLIParser() {
+        super(CONFIG);
+    }
+
+    public BootstrapCLIParser(Terminal terminal) {
+        super(CONFIG, terminal);
+    }
+
+    @Override
+    protected Command parse(String cmdName, CommandLine cli) throws Exception {
+        switch (cmdName.toLowerCase(Locale.ROOT)) {
+            case Start.NAME:
+                return Start.parse(terminal, cli);
+            case Version.NAME:
+                return Version.parse(terminal, cli);
+            default:
+                assert false : "should never get here, if the user enters an unknown command, an error message should be shown before parse is called";
+                return null;
+        }
+    }
+
+    static class Version extends CliTool.Command {
+
+        private static final String NAME = "version";
+
+        private static final CliToolConfig.Cmd CMD = cmd(NAME, Version.class).build();
+
+        public static Command parse(Terminal terminal, CommandLine cli) {
+            return new Version(terminal);
+        }
+
+        public Version(Terminal terminal) {
+            super(terminal);
+        }
+
+        @Override
+        public ExitStatus execute(Settings settings, Environment env) throws Exception {
+            terminal.println("Version: %s, Build: %s/%s, JVM: %s", org.elasticsearch.Version.CURRENT, Build.CURRENT.hashShort(), Build.CURRENT.timestamp(), JvmInfo.jvmInfo().version());
+            return ExitStatus.OK_AND_EXIT;
+        }
+    }
+
+    static class Start extends CliTool.Command {
+
+        private static final String NAME = "start";
+
+        private static final CliToolConfig.Cmd CMD = cmd(NAME, Start.class)
+                .options(
+                        optionBuilder("d", "daemonize").hasArg(false).required(false),
+                        optionBuilder("p", "pidfile").hasArg(true).required(false),
+                        optionBuilder("V", "version").hasArg(false).required(false),
```

Comment:
I would say that we should have `-v` as verbose as it stands now and just support `--version` for the version parameter and not have a short form for it (if thats possible). If not, I think this is a good solution. `-v` is fairly universal as a flag for verbose.